tools: Add script to install "upip" package manager.

pfalcon opened the pull request:
upip is a simple and light-weight package manager for MicroPython modules,
offering subset of pip functionality. upip is part of micropython-lib
project: https://github.com/micropython/micropython-lib/tree/master/upip
This script bootstraps upip by downloading and unpacking it directly from
PyPI repository, with all other packages to be installed with upip itself.

pfalcon commented:
So, I'd like everyone start taking advantage of this work, to fix issues https://github.com/micropython/micropython/issues/839 and having good step towards completely self-hosted package manager (current iteration isn't, it depends on running under unix with wget available to handle HTTPS).

More info about upip: http://forum.micropython.org/viewtopic.php?f=5&t=524 , https://github.com/micropython/micropython-lib/issues/19

pfalcon commented:
@stinos reports testing success at http://forum.micropython.org/viewtopic.php?f=5&t=524&start=10#p3978

dpgeorge commented on tools/bootstrap_upip.sh:
```diff
@@ -0,0 +1,13 @@
+# All the other packages can be installed using it.
+
+# Remove any stale old version
+rm -rf micropython-upip-*
```
I think it's too dangerous to have such a command in a bash script that people will blindly run anywhere on their system.  Can we wrap it in a check that such a directory exists?

pfalcon replied:
Well, we can do all that stuff in /tmp or $TMPDIR.

dpgeorge replied:
I think that would be better than doing it in the current directory.  Or force the user to provide a path (eg . or /tmp) to extract the tgz.

dpgeorge commented:
I get: `ImportError: no module named 'struct'` :(

pfalcon commented:
> I get: ImportError: no module named 'struct' :(

Oops, need to update for ``ustruct`` change.

pfalcon commented:
Updated upip and other affected modules.

Commit:
tools: Add script to install "upip" package manager.

upip is a simple and light-weight package manager for MicroPython modules,
offering subset of pip functionality. upip is part of micropython-lib
project: https://github.com/micropython/micropython-lib/tree/master/upip
This script bootstraps upip by downloading and unpacking it directly from
PyPI repository, with all other packages to be installed with upip itself.

pfalcon commented:
Updated to do stuff in /tmp.
